contracts/interfaces/IBalancerV2Vault.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/*
+    Subset of Balancer's IVault interface for the Enzyme Protocol.
+
+    For more info see: @balancer-labs/v2-vault/contracts/interfaces/IVault.sol
+*/
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./IBalancerV2Asset.sol";
+
+/// @title IBalancerVault Interface
+/// @author JustSomeGeeks Hackathon Team <https://github.com/justsomegeeks>
+/// @notice An Enzyme Balancer V2 Vault Interface
+/// @dev Provides a subset of the functionality copied from and provided by Balancer
+/// @dev Full documentation of structs and functions can be found in Balancer's code
+interface IBalancerV2Vault {
+    function joinPool(
+        bytes32 poolId,
+        address sender,
+        address recipient,
+        JoinPoolRequest memory request
+    ) external payable;
+
+    struct JoinPoolRequest {
+        IBalancerV2Asset[] assets;
+        uint256[] maxAmountsIn;
+        bytes userData;
+        bool fromInternalBalance;
+    }
+
+    function exitPool(
+        bytes32 poolId,
+        address sender,
+        address payable recipient,
+        ExitPoolRequest memory request
+    ) external;
+
+    struct ExitPoolRequest {
+        IBalancerV2Asset[] assets;
+        uint256[] minAmountsOut;
+        bytes userData;
+        bool toInternalBalance;
+    }
+
+    event PoolBalanceChanged(
```

Comment:
Do we need events in our interface?

Reply:
Good question. I'm not sure yet TBH. If not, let's delete them.